nifi-nar-bundles/nifi-influxdb-bundle/nifi-influxdb-processors/src/main/java/org/apache/nifi/processors/influxdb/ExecuteInfluxDBQuery.java

```diff
@@ -0,0 +1,258 @@
+        tempDescriptors.add(PASSWORD);
+        tempDescriptors.add(CHARSET);
+        propertyDescriptors = Collections.unmodifiableList(tempDescriptors);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        super.onScheduled(context);
+        // Either input connection or scheduled query is required
+        if ( ! context.getProperty(INFLUX_DB_QUERY).isSet()
+           && ! context.hasIncomingConnection() ) {
+            String error = "The InfluxDB Query processor requires input connection or scheduled InfluxDB query";
+            getLogger().error(error);
+            throw new ProcessException(error);
+        }
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+
+        String query = null;
+        String database = null;
+        TimeUnit queryResultTimeunit = null;
+        Charset charset = null;
+        FlowFile outgoingFlowFile = null;
+
+        // If there are incoming connections, prepare query params from flow file
+        if ( context.hasIncomingConnection() ) {
+            FlowFile incomingFlowFile = session.get();
+
+            if ( incomingFlowFile == null && context.hasNonLoopConnection() ) {
+                return;
+            }
+
+            if ( incomingFlowFile.getSize() == 0) {
+                getLogger().error("Empty query");
+                incomingFlowFile = session.putAttribute(incomingFlowFile, INFLUX_DB_ERROR_MESSAGE, "Empty query size is " + incomingFlowFile.getSize());
+                session.transfer(incomingFlowFile, REL_FAILURE);
+                return;
+            }
+
+            charset = Charset.forName(context.getProperty(CHARSET).evaluateAttributeExpressions(incomingFlowFile).getValue());
+            try {
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                session.exportTo(incomingFlowFile, baos);
+                baos.close();
+                query = new String(baos.toByteArray(), charset);
+            } catch(IOException ioe) {
+                getLogger().error("Exception while reading from FlowFile " + ioe.getLocalizedMessage(), ioe);
+                throw new ProcessException(ioe);
+            }
+
+            database = context.getProperty(DB_NAME).evaluateAttributeExpressions(incomingFlowFile).getValue();
+            queryResultTimeunit = TimeUnit.valueOf(context.getProperty(INFLUX_DB_QUERY_RESULT_TIMEUNIT).evaluateAttributeExpressions(incomingFlowFile).getValue());
+
+            outgoingFlowFile = incomingFlowFile;
+        } else {
+            charset = Charset.forName(context.getProperty(CHARSET).evaluateAttributeExpressions().getValue());
```

Comment:
These should be evaluated against the flowfile's attributes with `evaluateAttributeExpressions(FlowFile)`. I think it will work if the flowfile is null. A test against that wouldn't hurt just to be safe as well.

Reply:
Updated to use flow file for EL.